Fix layout issues and update markup

cristianfalcone opened the pull request:
Fixes #4 
Fixes #31 
Fixes #32

cristianfalcone committed:
Update Create VCH wizard markup

 - Update markup and styles
 - Use form controls in a grid
 - Use Clarity's default styles where possible

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/storage-capacity/storage-capacity.html:
```diff
@@ -1,91 +1,194 @@
+
       </div>
+      <div class="col-xs-2"></div>
+      <div class="col-xs"></div>
```
What are these empty divs for?

cristianfalcone replied:
This is to set space for previous auto layout columns, so form controls align to the same column. I will add a markup comment per occurrence to explain this.

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html:
```diff
@@ -1,136 +1,348 @@
+          <span class="tooltip-content">{{ formErrMessage }}</span>
+
+          <small class="text-muted">
+            Max: {{ resourceLimits.memory.maxUsage || '-' }} MB, Min: 1 MB
```
The Min value shouldn't be defined in the markup it should be same as Max. I know it was already here but if you can update it that would be great.

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html:
```diff
@@ -1,136 +1,348 @@
+          <span class="tooltip-content">{{ formErrMessage }}</span>
+
+          <small class="text-muted">
+            Max: {{ resourceLimits.cpu.unreservedForPool || '-' }} MHz, Min: 1 MHz
```
Same comment as previous. Min value needs to not be defined in template.

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html:
```diff
@@ -1,136 +1,348 @@
+          <span class="tooltip-content">{{ formErrMessage }}</span>
+
+          <small class="text-muted">
+            Max: {{ resourceLimits.memory.unreservedForPool || '-' }} MB, Min: 1 MB
```
Min value needs to not be defined in template.

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/compute-capacity/compute-capacity.html:
```diff
@@ -1,136 +1,348 @@
+          <span class="tooltip-content">{{ formErrMessage }}</span>
+
+          <small class="text-muted">
+            Max: {{ resourceLimits.memory.maxUsage || '-' }} MB, Min: 1 MB
```
Min value needs to not be defined in template.

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/create-vch-wizard.component.html:
```diff
@@ -1,7 +1,18 @@
+<clr-wizard #wizardlg
+            clrWizardSize="lg"
+            class="clr-wizard--inline clr-wizard--no-shadow"
+            style="height: 300px"
```
Is there any where we can add this into a css class?

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.html:
```diff
@@ -1,16 +1,48 @@
+
+          <input id="nameInput"
+                 class="form-control"
+                 size="50"
```
This seems rather arbitrary. Using 'size' can be inconsistent across browsers. Maybe we can use a CSS equivalent size such as 'ch'. Add to style sheet if feasible. But either way let's validate against the mockup and check cross browser.

cristianfalcone replied:
Sorry. This shouldn't be here. The input's parent with .form-group and .row classes, and input .form-control class make the input width adjust to the grid.

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/name/vch-creation-wizard-name.scss:
```diff
@@ -1,14 +1,4 @@
-
-.smaller-p-body-text {
-  font-size: 12px;
+:host /deep/ clr-signpost .signpost-action {
```
let's verify current browser support for /deep/ selector

cristianfalcone replied:
/deep/ is interpreted by angular cli in build time and rendered with standard css selectors. It allows us to by-pass child components styles encapsulation. Clarity is adding margins to all buttons that are form child elements. Even buttons rendered by Clarity's Signpost components. That's why we need this. I will replace this with its alias (::ng-deep) to make it more specific.

jak-atx commented on ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.scss:
```diff
@@ -1,60 +1,9 @@
-    color: red;
-    margin: 0;
-  }
+:host /deep/ clr-signpost .signpost-action {
```
verify current browser support for /deep/ selector

jak-atx reviewed (commented):
Great cleanup, thanks. Just a few small things noted in the comments. And please make sure to test cross browsers and list all browsers/platforms tested in commit msg. Thanks.

cristianfalcone committed:
Address comments from @jak-atx

 - Remove size attrs and format markup
 - All the wizard steps layout changes were tested in:
    - Latest Google Chrome version (MacOS)
    - Latest Firefox version (MacOS)
    - Internet Explorer 11 (Windows)

cristianfalcone commented:
@jak-atx, the wizard modal size is broken in IE11, I think @jooskim is taking care of this in another issue, but the steps layout looks good in IE11 too.

jooskim commented:
Thanks @cristianfalcone, I'm addressing this bug in issue #28

jak-atx reviewed (approved):
Looks good. Be sure to rebase fix any conflicts before merge.

cristianfalcone commented:
@jak-atx Done with rebase. Which strategy should I choose to merge?

jak-atx commented:
@cristianfalcone once rebased go ahead and merge using 'squash and merge'
